python/demo/poisson/demo_poisson.py

```diff
@@ -134,10 +141,19 @@
 # immediately above. The definition of the Dirichlet boundary condition
 # then looks as follows: ::
 
+# Create a function that returns True if the point is on the boundary
+
+
+def marker(x):
+    left = np.isclose(x[0], 0.0)
+    right = np.isclose(x[0], 2.0)
+    return np.logical_or(left, right)
+
+
```

Comment:
I like the lambda, and I like introducing users to it - it emphasises the data-centric design that distinguishes DOLFINx.

Reply:
Sounds good. I reverted this change in https://github.com/FEniCS/dolfinx/pull/1979/commits/b48e86cbdefc812f07b0e2f131a7d799853168fe